server/src/internalClusterTest/java/org/elasticsearch/search/ccs/CrossClusterSearchIT.java

```diff
@@ -0,0 +1,189 @@
+            .setSettings(Settings.builder().put("index.routing.allocation.require._name", remoteNode)
+                .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0).build()));
+        indexDocs(client("cluster_a"), "prod");
+        SearchListenerPlugin.blockQueryPhase();
+        try {
+            PlainActionFuture<SearchResponse> future = new PlainActionFuture<>();
+            SearchRequest searchRequest = new SearchRequest("demo", "cluster_a:prod");
+            searchRequest.allowPartialSearchResults(false);
+            searchRequest.setCcsMinimizeRoundtrips(false);
+            searchRequest.source(new SearchSourceBuilder().query(new MatchAllQueryBuilder()).size(1000));
+            client(LOCAL_CLUSTER).search(searchRequest, future);
+            SearchListenerPlugin.waitSearchStarted();
+            disconnectFromRemoteClusters();
+            assertBusy(() -> assertTrue(future.isDone()));
+            configureAndConnectsToRemoteClusters();
+        } finally {
+            SearchListenerPlugin.allowQueryPhase();
+        }
+    }
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins(String clusterAlias) {
+        if (clusterAlias.equals(LOCAL_CLUSTER)) {
+            return super.nodePlugins(clusterAlias);
+        } else {
+            final List<Class<? extends Plugin>> plugins = new ArrayList<>(super.nodePlugins(clusterAlias));
+            plugins.add(SearchListenerPlugin.class);
+            return plugins;
+        }
+    }
+
+    @Before
+    public void resetSearchListenerPlugin() throws Exception {
+        SearchListenerPlugin.reset();
+    }
+
+    public static class SearchListenerPlugin extends Plugin {
+        private static final AtomicReference<CountDownLatch> startedLatch = new AtomicReference<>();
+        private static final AtomicReference<CountDownLatch> queryLatch = new AtomicReference<>();
+
+        static void reset() {
+            startedLatch.set(new CountDownLatch(1));
+        }
+
+        static void blockQueryPhase() {
+            queryLatch.set(new CountDownLatch(1));
+        }
+
+        static void allowQueryPhase() {
+            final CountDownLatch latch = queryLatch.get();
+            if (latch != null) {
+                latch.countDown();
+            }
+        }
+
+        static void waitSearchStarted() throws InterruptedException {
+            startedLatch.get().await();
+        }
+
+        @Override
+        public void onIndexModule(IndexModule indexModule) {
+            indexModule.addSearchOperationListener(new SearchOperationListener() {
+                @Override
+                public void onPreQueryPhase(SearchContext searchContext) {
+                    startedLatch.get().countDown();
+                    final CountDownLatch latch = queryLatch.get();
+                    if (latch != null) {
+                        try {
+                            latch.await();
```

Comment:
NIT: Maybe put a timeout on these `.await` here and above and `assertTrue` to make test failures a little more obvious?

Reply:
Sure, adjusted in https://github.com/elastic/elasticsearch/pull/65697/commits/031bf821152fb07cc37e7ff63b89d298aa491ae3